src/Microsoft.Graph.Auth/Extensions/IClientApplicationBaseExtensions.cs

```diff
@@ -41,10 +41,13 @@ internal static async Task<AuthenticationResult> GetAccessTokenSilentAsync(this
 
             try
             {
-                return await clientApplication.AcquireTokenSilent(msalAuthProviderOption.Scopes, account)
-                        .WithAuthority(clientApplication.Authority)
-                        .WithForceRefresh(msalAuthProviderOption.ForceRefresh)
-                        .ExecuteAsync();
+                AcquireTokenSilentParameterBuilder tokenSilentBuilder = clientApplication.AcquireTokenSilent(msalAuthProviderOption.Scopes, account)
+                    .WithForceRefresh(msalAuthProviderOption.ForceRefresh);
+
+                if (ContainsWellKnownTenantName(clientApplication.Authority))
```

Comment:
Isn't this the opposite of what we want?  e.g. We don't want to call WithAuthority(...)  if the Authority is one of the well known names

Reply:
Good catch. I omitted `!` which my last commit.